common/static/common/js/discussion/utils.js

```diff
@@ -444,19 +475,45 @@
             }
         };
 
-        DiscussionUtil.abbreviateHTML = function(html, minLength) {
+        DiscussionUtil.convertMath = function(element) {
+            edx.HtmlUtils.setHtml(
+                element,
+                DiscussionUtil.postMathJaxProcessor(DiscussionUtil.markdownWithHighlight(element.html()))
+            );
+        };
+
+        DiscussionUtil.abbreviateHTML = function(htmlSnippet, minLength) {
             var $result, imagesToReplace, truncated_text;
-            truncated_text = jQuery.truncate(html, {
+            truncated_text = edx.HtmlUtils.HTML(jQuery.truncate(htmlSnippet.toString(), {
                 length: minLength,
                 noBreaks: true,
                 ellipsis: gettext('…')
-            });
-            $result = $("<div>" + truncated_text + "</div>");
+            }));
+            $result = $(edx.HtmlUtils.joinHtml(
+                edx.HtmlUtils.HTML("<div>"),
+                truncated_text,
+                edx.HtmlUtils.HTML("</div>")
+            ).toString());
             imagesToReplace = $result.find("img:not(:first)");
             if (imagesToReplace.length > 0) {
-                $result.append("<p><em>Some images in this post have been omitted</em></p>");
+                edx.HtmlUtils.append(
+                    $result,
+                    edx.HtmlUtils.joinHtml(
```

Comment:
A few more instances of the HTML pattern Robert and I commented on above down in this file.